src/psyclone/tests/parse/file_info_test.py

```diff
@@ -404,10 +403,37 @@ def test_file_info_cachefile_not_accessible(tmpdir):
     with open(filename, "w", encoding="utf-8") as fout:
         fout.write(SOURCE_DUMMY)
 
-    file_info: FileInfo = FileInfo(filename, use_caching=True)
+    file_info: FileInfo = FileInfo(filename, cache_active=True)
 
     # Set buggy cache file
-    file_info._filepath_cache = "/I_DONT_EXIST/FILE/cache.psycache"
+    file_info._cache_path = "/I_DONT_EXIST/FILE/cache.psycache"
+
+    source_code = file_info.get_source_code(verbose=True)
+    assert source_code == SOURCE_DUMMY
+
+    psyir_node = file_info.get_psyir(verbose=True)
+    assert isinstance(psyir_node, Node)
+
+
+def test_file_info_cachefile_not_writable(tmpdir):
+    '''
+    Check if cachefile is not writable.
+    This should not raise any errors!
+    '''
+
+    filename = os.path.join(tmpdir, "testfile_e.f90")
+
+    try:
+        os.remove(filename)
+    except FileNotFoundError:
```

Comment:
This can be improved by using the `tmp_path` fixture instead of `tmpdir` - that will ensure that the directory this test is given is unique to it so the file can't exist.